router/core/graphql_prehandler.go

```diff
@@ -114,14 +131,115 @@ func (h *PreHandler) Handler(next http.Handler) http.Handler {
 		buf := pool.GetBytesBuffer()
 		defer pool.PutBytesBuffer(buf)
 
-		body, err := h.parser.ReadBody(r.Context(), buf, r.Body)
+		body, err := h.parser.ReadBody(buf, r.Body)
 		if err != nil {
 			finalErr = err
 			requestLogger.Error(err.Error())
 			writeRequestErrors(r.Context(), http.StatusBadRequest, graphql.RequestErrorsFromError(err), w, requestLogger)
 			return
 		}
 
+		/**
+		* Parse the operation
+		 */
+
+		engineParseCtx, engineParseSpan := h.tracer.Start(r.Context(), "Operation - Parse",
+			trace.WithSpanKind(trace.SpanKindServer),
+		)
+
+		parser, err := h.parser.NewParser(body)
```

Comment:
Does this create one parser per request without being able to re-use it?
If yes, is there a way to re-use parsers more?
If parsers are already being re-used, close this comment.

Reply:
I'll try to improve the naming. Only one parser is created but we create an OperationKit per request that yields the result of the parsed, normalized and validated result.